controllers/storagecluster/exporter.go

```diff
@@ -108,6 +110,14 @@ func (r *StorageClusterReconciler) enableMetricsExporter(
 	if err != nil {
 		return err
 	}
+
+	// create a `ocs-metrics-exporter-ceph-auth` secret for metrics exporter
+	// create a cephclient and it will create the secret
+	err = r.createMetricsExporterCephClient(instance)
```

Comment:
We dont have the deletion code for any of these resources, so @rewantsoni  we are fine for now

Reply:
its the pattern to follow, create something in Ensure Create and delete it in EnsureDelete to avoid cleanup problems.

Reply:
Discard my above comment as the code is moved

Reply:
Testing:

```
rider@192:~$ oc get cephclient  ocs-metrics-exporter-ceph-auth 
NAME                             PHASE   AGE
ocs-metrics-exporter-ceph-auth   Ready   27m
rider@192:~$ 
rider@192:~$ 
rider@192:~$ oc delete storagecluster ocs-storagecluster
storagecluster.ocs.openshift.io "ocs-storagecluster" deleted
^Z
[1]+  Stopped                 oc delete storagecluster ocs-storagecluster

I didn't let it to complete, because there were data in the cluster,etc

rider@192:~$ 
rider@192:~$ oc get cephclient  ocs-metrics-exporter-ceph-auth 
Error from server (NotFound): cephclients.ceph.rook.io "ocs-metrics-exporter-ceph-auth" not found
rider@192:~$ 


```